boards/arm/rcar_h3ulcb/doc/rcar_h3ulcb.rst

```diff
@@ -0,0 +1,169 @@
+   :width: 460px
+   :align: center
+   :height: 288px
+   :alt: R-Car starter kit
+
+.. Note:: The H3ULCB board can be plugged on a Renesas Kingfisher Infotainment daughter board through COM Express connector in order to physically access more I/O. CAUTION : In this case, power supply is managed by the daughter board.
+
+More information about the board can be found at `Renesas R-Car Starter Kit website`_.
+
+Hardware
+********
+
+Hardware capabilities for the H3ULCB for can be found on the `eLinux H3SK page`_ of the board.
+
+.. figure:: img/rcar_h3ulcb_features.jpg
+   :width: 286px
+   :align: center
+   :height: 280px
+   :alt: R-Car starter kit features
+
+.. Note:: Zephyr will be booted on the CR7 processor provided for RTOS purpose.
+
+More information about the SoC that equips the board can be found here :
+
+- `Renesas R-Car H3 chip`_
+
+Supported Features
+==================
+
+Here is the current supported features when running Zephyr Project on the R-Car ULCB CR7:
+
++-----------+------------------------------+
+| Interface | Driver/components            |
++===========+==============================+
+| CLOCK     | clock_control                |
++-----------+------------------------------+
+| GPIO      | gpio                         |
++-----------+------------------------------+
+| GIC       | GIC 400 interrupt controller |
++-----------+------------------------------+
+
+It's also currently possible to write on the ram console.
+
+More features will be supported soon.
+
+
+Connections and IOs
+===================
+
+H3ULCB Board :
+------------------
+
+Here are official IOs figures from eLinux for H3ULCB board :
+
+`H3SK top view`_
+
+`H3SK bottom view`_
+
+Kingfisher Infotainment daughter board :
+----------------------------------------
+
+When connected to Kingfisher Infotainment board through COMExpress connector, the board is exposing much more IOs.
+
+Here are official IOs figures from eLinux for Kingfisher Infotainment board :
+
+`Kingfisher top view`_
+
+`Kingfisher bottom view`_
+
+Programming and Debugging
+*************************
+
+The Cortex®-R7 of rcar_h3ulcb board needs to be started by the Cortex®-A cores. Cortex®-A cores are responsible to load the Cortex®-R7 binary application into the RAM, and get the Cortex®-R7 out of reset. The Cortex®-A can currently perform these steps at bootloader level.
+
+Building
+========
+
+Applications for the ``rcar_h3ulcb`` board configuration can be built in the usual way (see :ref:`build_an_application` for more details).
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/hello_world
+   :board: rcar_h3ulcb
+   :goals: build
+
+Debugging
+=========
+You can debug an application using OpenOCD and GDB. The Solution proposed below is using a OpenOCD custom version that support R-Car ULCB boards Cortex®-R7.
+
+Get Renesas ready OpenOCD version
+---------------------------------
+
+.. code-block:: bash
+
+	git clone --branch renesas https://github.com/iotbzh/openocd.git
```

Comment:
Maybe you could consider upstreaming those patches or merging them into [our OpenOCD fork](https://github.com/zephyrproject-rtos/openocd), which includes support for Zephyr thread awareness. Otherwise users of this board won't get those extra features.

Reply:
Hi, Zephyr awareness is included in my own fork, i'm still thinking about either cleaning and contributing to zephyr openocd, or upstream openocd, or finding another tool.